Sukanta_Roy/progress.md

```diff
@@ -47,30 +47,49 @@ Week 2 Retrospect:
 In week 2, due to my final exams, I was not able to work on my project. But as the plan was to finalize the sample question list for the new template, I made three templates with sample questions to choose from. The plan is to create a template that will focus on one aspect only (e.g. Team dynamics, Performance based questions with no open ended essay questions etc.). Though further discussion is needed with my mentors and Organization admins to finalize one template that will add the most value to teammates and the instructors. Hoping to do it after May 30th, as Prof. Damith will be available after that.
 
 Week 3 Plan:
-- [ ] Finish implementing the backend part of [MCQ questions: support attaching weights to options](https://github.com/TEAMMATES/teammates/issues/2776)
+- [x] Finish implementing the backend part of [MCQ questions: support attaching weights to options](https://github.com/TEAMMATES/teammates/issues/2776)
 - [ ] Implement the frontend part of [MCQ questions: support attaching weights to options](https://github.com/TEAMMATES/teammates/issues/2776)
 	- [ ] Update question template to add the new UI for MCQ weights
 	- [ ] Add JS functionalities
 	- [ ] Add necessary JS tests
 	- [ ] Add frontend UI tests
 
+Week 3 Work:
+- [x] Merged: [Instructor: Edit session: Rubric question: Add front-end validation for empty weights](https://github.com/TEAMMATES/teammates/issues/8631)
+- [x] Merged: [Rubric question add/edit action test: Add missing parameter and additional tests](https://github.com/TEAMMATES/teammates/issues/8855)
+- [x] Finished adding backend implementation for MCQ weights feature in [Instructor: MCQ questions (single answer): support attaching weights to options #2776](https://github.com/TEAMMATES/teammates/pull/8860)
+
+Week 3 Retrospect:
+In this week, created the pull request for MCQ weights and added backend for MCQ weights. It took longer than expected to add tests for backend as I am relatively new to Unit testing. Apart from that, no major setbacks.
```

Comment:
I believe you have learnt a lot about how to write unit test and you are doing good actually :)